lib/spack/spack/audit.py

```diff
@@ -297,6 +297,30 @@ def _check_build_test_callbacks(pkgs, error_cls):
     return errors
 
 
+@package_directives
+def _check_legacy_attributes(pkgs, error_cls):
+    """Ensure legacy attributes are not defined in package classes."""
```

Comment:
This is wrong, we marked legacy attributes those that _can_ be used in package classes, if people code the package as a single class. I'll have a look at your issue going to the end of the week.

Reply:
Fair enough. Withdrawing PR.